descartes_light/solvers/include/descartes_light/solvers/ompl/impl/descartes_space.hpp

```diff
@@ -0,0 +1,424 @@
+  if ((sample_up || rung == 0) && rung != ladder_rungs.size() - 1)
+    new_rung = rung + 1;
+  else
+    new_rung = rung - 1;
+  // Find new index on this new sampled rung
+  std::size_t new_idx = static_cast<std::size_t>(rng_.uniformInt(0, static_cast<int>(ladder_rungs[new_rung].size() - 1)));
+  // Create vertex with the newly sampled rung and index
+  std::pair<long unsigned int, long unsigned int> new_vertex(new_rung, new_idx);
+  // Store new vertex to the returned state
+  state->as<typename descartes_light::DescartesStateSpace<FloatType>::StateType>()->vertex = new_vertex;
+  // Ensure the new sample is within the ladder graph (it shouldn't be possible to be outside)
+  space_->enforceBounds(state);
+}
+
+template <typename FloatType>
+void descartes_light::DescartesStateSampler<FloatType>::sampleGaussian(ompl::base::State *state, const ompl::base::State *mean, const double stdDev)
+{
+  // Get the ladder rung structure to have knowledge of the available sample space
+  std::vector<std::vector<VertexDesc<FloatType>>> ladder_rungs =
+      space_->as<descartes_light::DescartesStateSpace<FloatType>>()->getLadderRungs();
+  // Find the distance between rungs to account for high stddev
+  const double rung_2_rung_dist = space_->as<descartes_light::DescartesStateSpace<FloatType>>()->getRungToRungDist();
+  // Find the rung we are trying to sample near
+  double rung = static_cast<double>(mean->as<typename descartes_light::DescartesStateSpace<FloatType>::StateType>()->vertex.first);
+  // Determine new rung to be sampled on
+  long unsigned int new_rung = static_cast<long unsigned int>(floor(rng_.gaussian(rung, stdDev / rung_2_rung_dist) + 0.5));
+  // Make sure we are inside the state space
+  if (new_rung >= ladder_rungs.size())
+    new_rung = ladder_rungs.size() - 1;
+  // Find new random index on this new sampled rung
+  std::size_t new_idx = static_cast<std::size_t>(rng_.uniformInt(0, static_cast<int>(ladder_rungs[new_rung].size() - 1)));
+  // Create vertex with the newly sampled rung and index
+  std::pair<long unsigned int, long unsigned int> new_vertex(new_rung, new_idx);
+  // Store new vertex to the returned state
+  state->as<typename descartes_light::DescartesStateSpace<FloatType>::StateType>()->vertex = new_vertex;
+  // Ensure the new sample is within the ladder graph (it shouldn't be possible to be outside)
+  space_->enforceBounds(state);
+}
+
+template <typename FloatType>
+bool descartes_light::DescartesStateSpace<FloatType>::isDiscrete() const
+{
+  return true; // This is a discrete state space
+}
+
+template <typename FloatType>
+unsigned int descartes_light::DescartesStateSpace<FloatType>::getDimension() const
+{
+  return 2; // This state space is 2-dimensional
+}
+
+template <typename FloatType>
+double descartes_light::DescartesStateSpace<FloatType>::getMaximumExtent() const
+{
+  return rung_to_rung_dist_ * static_cast<double>(ladder_rungs_.size() + 1);
```

Comment:
Why `ladder_rungs_.size() + 1`? There are only `ladder_rungs_.size() - 1` rung connections

Reply:
Yes, `rung_to_rung_dist_ * static_cast<double>(ladder_rungs_.size() - 1)` would make the most sense. Initially I was both trying to account for index difference potentials and had a typo. There will be no index difference present in the maximum extent case because both the last and first rung only have 1 vertex since they are artificially added start and end states.